dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Guest/SetShippingMethodsOnCartTest.php

```diff
@@ -42,100 +48,437 @@ protected function setUp()
         $this->quoteResource = $objectManager->get(QuoteResource::class);
         $this->quoteFactory = $objectManager->get(QuoteFactory::class);
         $this->quoteIdToMaskedId = $objectManager->get(QuoteIdToMaskedQuoteIdInterface::class);
+        $this->quoteAddress = $objectManager->get(QuoteAddress::class);
     }
 
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_virtual_product_saved.php
+     */
     public function testShippingMethodWithVirtualProduct()
     {
-        $this->markTestIncomplete('https://github.com/magento/graphql-ce/issues/422');
+        $methodCode = 'flatrate';
+        $carrierCode = 'flatrate';
+        $reservedOrderId = 'test_order_with_virtual_product_without_address';
+
+        $maskedQuoteId = $this->getMaskedQuoteIdByReversedQuoteId($reservedOrderId);
+        $quoteAddressId = $this->getQuoteAddressIdByReversedQuoteId($reservedOrderId);
+
+        $query = $this->prepareMutationQuery(
+            $maskedQuoteId,
+            $methodCode,
+            $carrierCode,
+            $quoteAddressId
+        );
+
+        self::expectException(\Exception::class);
+        self::expectExceptionMessage('Carrier with such method not found: ' . $methodCode . ', ' . $methodCode . '');
+        $this->graphQlQuery($query);
     }
 
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_simple_product_saved.php
+     */
     public function testShippingMethodWithSimpleProduct()
     {
-        $this->markTestIncomplete('https://github.com/magento/graphql-ce/issues/422');
+        $methodCode = 'flatrate';
+        $carrierCode = 'flatrate';
+        $reservedOrderId = 'test_order_with_simple_product_without_address';
+
+        $maskedQuoteId = $this->getMaskedQuoteIdByReversedQuoteId($reservedOrderId);
+        $quoteAddressId = $this->getQuoteAddressIdByReversedQuoteId($reservedOrderId);
```

Comment:
Looks like we can reuse `Magento/Checkout/_files/quote_with_address_saved.php` instead of creating `getQuoteAddressIdByReversedQuoteId`
Please check it

Reply:
Yes, we can reuse `Magento/Checkout/_files/quote_with_address_saved.php`.  Modified test case, please review it.